column2Vec/reaserch/column2Vec_re.py

```diff
@@ -0,0 +1,192 @@
+    :param key: key for cache
+    """
+    emb1 = function(column1, model, key)
+    emb2 = function(column1, model, key + "s1")
+    emb3 = function(column1, model, key + "s2")
+    first_second = cosine_sim(emb3, emb2)
+    whole_second = cosine_sim(emb1, emb2)
+    whole_first = cosine_sim(emb3, emb1)
+    if first_second != 1 or whole_second != 1 or whole_first != 1:
+        return False, f"For {key}:\n is not stable\n"
+    return True, f"For {key}: is stable\n"
+
+
+def similarity_test(embeddings: dict) -> dict:
+    """
+    Compute similarity of columns based on embeddings similarity/distance. It should be 1 for same columns.
+    Result from this function will be used in generate_report for counting similar and not similar score.
+    :param embeddings: embeddings of columns
+    """
+    keys = list(embeddings.keys())
+    res = dict()
+    # Initialize nested dictionaries for each function
+    for function in embeddings[list(embeddings.keys())[0]].keys():
+        res[function] = dict()
+
+    for column in embeddings.keys():
+        for function in embeddings[column].keys():
+            for column2 in keys:
+                sim = cosine_sim(embeddings[column][function], embeddings[column2][function])
+                res[function][column, column2] = sim
+                # res[function][column2, column] = sim
+        # keys.remove(column)
+    return res
+
+
+def time_test(column1: pd.Series, function, key: str) -> (bool, float):
+    """
+    Computes how much time it takes to compute embeddings for column using specific function.
+    Cache are not used for this test
+    :param column1: column to test
+    :param function: function to use
+    :param key: key for cache
+    """
+    # cache.set_file("no_file.txt")
+    # cache.clear_cache()
+    cache.off()
+    start = time.time()
+    function(column1, model, key)
+    end = time.time()
+    cache.on()
+    return True, (end - start)
+
+
+def partial_column_test(column1: pd.Series, function, key: str) -> (bool, str):
+    """
+    Test if column with only part of data is same/similar as column with whole data
+    :param column1: column to test
+    :param function: function to use
+    :param key: key for cache
+    :return: tuple of bool and str
+    """
+    length = column1.size
+    first_half = column1.head(int(length / 2))
+    second_half = column1.tail(int(length / 2))
+
+    emb1 = function(column1, model, key)
+    emb2 = function(first_half, model, key + str(1))
+    emb3 = function(second_half, model, key + str(2))
+    first_second = cosine_sim(emb3, emb2)
+    whole_second = cosine_sim(emb1, emb2)
+    whole_first = cosine_sim(emb3, emb1)
+    if first_second != 1 or whole_second != 1 or whole_first != 1:
+        # print((f"For {key}, {function}:\n whole column and first half: {whole_first}\n "
+        #        f"whole column and second half: {whole_second}\n first half"
+        #        f" and second half: {first_second}\n"))
```

Comment:
Commented out code either has to have a good explanation next to it or should not be there.

Reply:
Fixed